### PR TITLE
Categories refactor: UI

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: webapp/views/App/views/Designer/CategoryList/TableRow/store/state/state.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export const getInitData = A.prop(keys.initData)
+export const getOnSelect = A.prop(keys.onSelect)
+export const getPosition = A.prop(keys.position)
+export const getSelectedItemUuid = A.prop(keys.selectedItemUuid)
+export const isUnused = A.prop(keys.unused)
+
+export const isSelected = (state) => {
+  const selectedItemUuid = getSelectedItemUuid(state)
+  return selectedItemUuid && selectedItemUuid === Category.getUuid(getCategory(state))
+}
```
Comment (ramirobg94) [inline webapp/views/App/views/Designer/CategoryList/TableRow/store/state/state.js:50]: good I think that I did worst ahhahahaha

File: webapp/components/survey/CategorySelector/CategorySelector.scss [hunk 2]
```diff
@@ -4,3 +4,7 @@
   display: grid;
   grid-template-columns: 1fr repeat(2, 100px);
 }
+
+.panel-right__content .table {
+  height: 100%;
+}
```
[the remark this answers is not in the document]
Comment (SteRiccio) [reply webapp/components/survey/CategorySelector/CategorySelector.scss:10]: ok

File: webapp/views/App/views/Designer/CategoryList/CategoryList.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  const inCategoriesPath = Boolean(matchPath(pathname, appModuleUri(designerModules.categories)))
+
+  const canEdit = useAuthCanEditSurvey()
+
+  const gridTemplateColumns = [
+    '50px', // index
+    '1fr', // name
+    ...(canEdit ? ['repeat(2, 80px)'] : []), // select button
+    ...(canSelect ? ['80px'] : []), // error and warning badges
```
[the remark this answers is not in the document]
Comment (SteRiccio) [reply webapp/views/App/views/Designer/CategoryList/CategoryList.js:30]: right, I've switched the comments!

File: webapp/views/App/views/Designer/CategoryList/CategoryList.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.badge {
+  position: unset;
+}
```
[the remark this answers is not in the document]
Comment (SteRiccio) [reply webapp/views/App/views/Designer/CategoryList/CategoryList.scss:3]: right

**Comment by SteRiccio** [comment]
> Also, could you remove the Items component , since I guess it's not used anymore?

ok

File: webapp/components/survey/NodeDefDetails/NodeDefDetails.scss [hunk 5]
```diff
@@ -54,3 +54,15 @@
...
+.node-def-edit__container {
+  .panel-right__content {
+    .table {
+      height: 100%;
+
+      .badge {
+        position: inherit;
+      }
+    }
+  }
```
Comment (minotogna) [inline webapp/components/survey/NodeDefDetails/NodeDefDetails.scss:67]: can u move this after line 26?

Comment (minotogna) [reply webapp/components/survey/NodeDefDetails/NodeDefDetails.scss:67]: or even better .node-def-edit__container .panel-right__content

Comment (SteRiccio) [reply webapp/components/survey/NodeDefDetails/NodeDefDetails.scss:67]: sure

**Pull request merged**